packages/components/inputs/creatable-select-input/src/creatable-select-input.tsx

```diff
@@ -397,6 +321,158 @@ CreatableSelectInput.defaultProps = {
+            (({ inputValue }) =>
+              inputValue === ''
+                ? intl.formatMessage(messages.noOptionsMessageWithoutInputValue)
+                : intl.formatMessage(messages.noOptionsMessageWithInputValue, {
+                    inputValue,
+                  }))
+          }
+          onBlur={
+            typeof props.onBlur === 'function'
+              ? () => {
+                  const event = {
+                    target: {
+                      name: (() => {
+                        if (!props.isMulti) return props.name;
+                        // We append the ".0" to make Formik set the touched
+                        // state as an array instead of a boolean only.
+                        // Otherwise the shapes would clash on submission, as
+                        // Formik will create an array on submission anyways.
+                        return props.name ? `${props.name}.0` : undefined;
+                      })(),
+                    },
+                    persist: () => {},
+                  };
+                  props.onBlur && props.onBlur(event);
+                }
+              : undefined
+          }
+          onChange={(value, info) => {
+            // selectedOptions is either an array, or a single option, or null
+            // depending on whether we're in multi-mode or not (isMulti)
+
+            let newValue = value;
+
+            if (props.isMulti && !newValue) {
+              newValue = [];
+            }
+
+            props.onChange(
+              {
+                target: { name: props.name, value: newValue },
+                persist: () => {},
+              },
+              info
+            );
+          }}
+          onFocus={props.onFocus}
+          onInputChange={props.onInputChange}
+          options={props.options}
+          placeholder={placeholder}
+          tabIndex={props.tabIndex}
+          tabSelectsValue={props.tabSelectsValue}
+          value={props.value}
+          // Creatable props
+          allowCreateWhileLoading={props.allowCreateWhileLoading}
+          formatCreateLabel={
+            props.formatCreateLabel ||
+            ((inputValue) =>
+              intl.formatMessage(messages.createLabel, {
+                inputValue,
+              }))
+          }
+          isValidNewOption={props.isValidNewOption}
+          getNewOptionData={props.getNewOptionData}
+          onCreateOption={props.onCreateOption}
+          createOptionPosition={props.createOptionPosition}
+          //@ts-ignore
+          iconLeft={props.iconLeft}
+        />
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+CreatableSelectInput.displayName = 'CreatableSelectInput';
+
+// Both "true" and an empty array [] represent a touched state. The Boolean
+// conveniently handles both cases
+CreatableSelectInput.isTouched = (touched: unknown) => Boolean(touched);
+
+CreatableSelectInput.defaultProps = defaultProps;
+
 addStaticFields(CreatableSelectInput, {
```

Comment:
Can you check how the type declarations are after the package is built?

We might need to manually expose some of the static fields as types, so that TS can recognize them.

Reply:
I don't think we need to manually expose them.  See below how they are built. 
![image](https://user-images.githubusercontent.com/6726467/151137479-f4d84dae-3654-4816-8022-03fa3f700012.png)

Reply:
Yes but the spreaded components are not exposed. Anyway, we can iterate on this as a follow up maybe.

For now it's fine like this.

Reply:
Okay 👍🏽